File: src/main/java/hackerrank/search/IceCreamParlor.java
package hackerrank.search;

import java.util.HashMap;
import java.util.Map;

public class IceCreamParlor {

    static void whatFlavors(int[] cost, int money) {
        final int[] flavors = getFlavors(cost, money);
        System.out.println(flavors[0] + " " + flavors[1]);
    }

    static int[] getFlavors(int[] cost, int money) {
        final Map<Integer, Integer> indices = new HashMap<>();

        for (int i = 0; i < cost.length; i++) {
            if (indices.containsKey(money - cost[i]))
                return new int[]{indices.get(money - cost[i]) + 1, i + 1};

            indices.put(cost[i], i);
        }

        throw new IllegalStateException();
    }

}

File: src/main/kotlin/hackerrank/search/IceCreamParlor.kt
package hackerrank.search

fun whatFlavors(cost: Array<Int>, money: Int): Pair<Int, Int> {
    val indices = mutableMapOf<Int, Int>()

    cost.forEachIndexed { i, c ->
        if (indices.containsKey(money - c)) {
            return indices[money - c]!! + 1 to i + 1
        }

        indices[c] = i
    }

    throw IllegalStateException()
}
